src/ast/codegen_llvm.cpp

```diff
@@ -1273,14 +1273,32 @@ void CodegenLLVM::visit(Ternary &ternary)
 void CodegenLLVM::visit(FieldAccess &acc)
 {
   SizedType &type = acc.expr->type;
-  assert(type.type == Type::cast || type.type == Type::ctx);
+  assert(type.type == Type::cast || type.type == Type::ctx ||
+         type.type == Type::tuple);
   acc.expr->accept(*this);
 
   if (type.is_kfarg)
   {
     expr_ = b_.CreatKFuncArg(ctx_, acc.type, acc.field);
     return;
   }
+  else if (type.type == Type::tuple)
+  {
+    // Get offset to start of the indexed element
+    size_t offset = 0;
+    for (ssize_t i = 0; i < acc.index; ++i)
+      offset += type.tuple_elems[i].size;
+
+    Value *src = b_.CreateAdd(expr_, b_.getInt64(offset));
```

Comment:
I  wonder if llvm will be happier and codegen simpler if we emit  a struct for  each tuple. These  5 lines of codegen could then be replaced a single CreateGEP as llvm does the offset math.

Reply:
Ok, I'll take a look

Reply:
Done